service/orchestrator/orchestrator.go

```diff
@@ -311,3 +312,100 @@ func (s *Service) informHook(result *assessment.AssessmentResult, err error) {
+		return nil, status.Errorf(codes.Internal, "could not add certificate to the database: %v", err)
+	}
+
+	return &emptypb.Empty{}, nil
+}
+
+// GetCertificate implements method for getting a certificate, e.g. to show its state in the UI
+func (s *Service) GetCertificate(_ context.Context, req *orchestrator.GetCertificateRequest) (response *orchestrator.Certificate, err error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, orchestrator.ErrRequestIsNil.Error())
+	}
+	if req.CertificateId == "" {
+		return nil, status.Errorf(codes.NotFound, orchestrator.ErrCertIDIsMissing.Error())
+	}
+
+	response = new(orchestrator.Certificate)
+	err = s.storage.Get(response, "Id = ?", req.CertificateId)
+	if errors.Is(err, persistence.ErrRecordNotFound) {
+		return nil, status.Errorf(codes.NotFound, "certificate not found")
+	} else if err != nil {
+		return nil, status.Errorf(codes.Internal, "database error: %s", err)
+	}
+	return response, nil
+}
+
+// ListCertificates implements method for getting a certificate, e.g. to show its state in the UI
+func (s *Service) ListCertificates(_ context.Context, req *orchestrator.ListCertificatesRequest) (response *orchestrator.ListCertificatesResponse, err error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, orchestrator.ErrRequestIsNil.Error())
+	}
+
+	response = new(orchestrator.ListCertificatesResponse)
+	err = s.storage.List(&response.Certificates)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "database error: %s", err)
+	}
+	return response, nil
+}
+
+// UpdateCertificate implements method for updating an existing certificate
+func (s *Service) UpdateCertificate(_ context.Context, req *orchestrator.UpdateCertificateRequest) (response *orchestrator.Certificate, err error) {
+	if req.CertificateId == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "certificate id is empty")
+	}
+
+	if req.Certificate == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "certificate is empty")
+	}
+
+	count, err := s.storage.Count(req.Certificate, "Certificate_id = ?", req.CertificateId)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "database error: %s", err)
+	}
+
+	if count == 0 {
+		return nil, status.Error(codes.NotFound, "certificate not found")
+	}
+
+	response = req.Certificate
+	response.Id = req.CertificateId
+
+	err = s.storage.Save(response, "Id = ?", response.Id)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "database error: %v", err)
+	}
+	return
+}
+
+// RemoveCertificate implements method for removing a certificate
+func (s *Service) RemoveCertificate(_ context.Context, req *orchestrator.RemoveCertificateRequest) (response *emptypb.Empty, err error) {
+	if req.CertificateId == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "certificate id is empty")
+	}
+
+	err = s.storage.Delete(&orchestrator.Certificate{}, "Id = ?", req.CertificateId)
+	if errors.Is(err, persistence.ErrRecordNotFound) {
```

Comment:
Just thinking out loud here, could we somehow wrap this into a `handleOrNotFound` function or something? we repeat this pattern quite often (not just here)

Reply:
Good point to think about. Created issue #650

Reply:
Thanks, we can do in a seperate PR, resolving this.